### PR TITLE
fix(QueryBuilder): Filters with regex values can be edited in place

### PR DESCRIPTION
**Pull request by grafakus**
### ✨ Description

**Related issue(s):** solves https://github.com/grafana/explore-profiles/issues/206, relates to https://github.com/grafana/grafana/issues/93664

The issue stems from the fact that the `Select` component from the [Saga design system](https://developers.grafana.com/ui/canary/index.html?path=/story/forms-select--basic) does not initialize its input with the value passed as prop. 

### 📖 Summary of the changes

To circumvent the problem, this PR replaces the `Select` component by an HTML `input`. 

See the diff tab for specific comments.

### 🧪 How to test?

Manually, by checking this PR ; here's a short video:

https://github.com/user-attachments/assets/041f85eb-f0da-49b8-8566-f87aa6d9f423

### CODE REVIEW

File: src/shared/components/QueryBuilder/domain/helpers/__tests__/queryToFilters.spec.ts [hunk 1]
```diff
@@ -79,6 +79,28 @@ const cases: TestCase[] = [
       },
     ],
   ],
+  [
+    ' process_cpu:cpu:nanoseconds:cpu:nanoseconds{service_name="core-requests",free_text=~"one,two"}',
```
Comment (grafakus) [inline src/shared/components/QueryBuilder/domain/helpers/__tests__/queryToFilters.spec.ts:83]: Modified the code and added a unit test to account for values that could contains commas.

File: src/shared/components/QueryBuilder/domain/types.ts [hunk 2]
```diff
@@ -48,6 +48,7 @@ export enum FilterPartKind {
 export type Suggestion = {
   value: string;
   label: string;
+  description?: string;
```
Comment (grafakus) [inline src/shared/components/QueryBuilder/domain/types.ts:51]: Added a description for consistency with the native Grafana AdHoc filters (e.g. see [Explore Metrics](https://play.grafana.org/explore/metrics/trail?from=now-1h&to=now&var-ds=grafanacloud-demoinfra-prom&var-otel_resources=&var-filters=&refresh=&var-deployment_environment=&metricPrefix=all))

File: src/shared/components/QueryBuilder/infrastructure/operatorsRepository.ts [hunk 3]
```diff
@@ -3,10 +3,10 @@ import { Suggestions } from '../domain/types';
 class OperatorsRepository {
   async list(): Promise<Suggestions> {
     return [
-      { value: '=', label: '=' },
-      { value: '!=', label: '!=' },
-      { value: '=~', label: '=~' },
-      { value: '!~', label: '!~' },
+      { value: '=', label: '=', description: 'Equals' },
```
Comment (grafakus) [inline src/shared/components/QueryBuilder/infrastructure/operatorsRepository.ts:6]: image

File: src/shared/components/QueryBuilder/domain/helpers/__tests__/queryToFilters.spec.ts [hunk 4]
```diff
@@ -79,6 +79,28 @@ const cases: TestCase[] = [
...
+          value: '=~',
+        },
+        value: {
+          label: 'one,two',
+          value: 'one,two',
+        },
+      },
+    ],
+  ],
   // TODO: uncomment when we'll support the "in" operator
```
Comment (ifrost) [inline src/shared/components/QueryBuilder/domain/helpers/__tests__/queryToFilters.spec.ts:104]: Can this be removed now?

File: src/shared/components/QueryBuilder/ui/inputs/SingleEditionInput.tsx [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+    onChange({ value, label: value });
+  };
+
+  useEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.focus();
+    }
+  });
+
+  // we have to implement a customized version of the input because the <Input /> from the Saga Design system does not
```
Comment (ifrost) [inline src/shared/components/QueryBuilder/ui/inputs/SingleEditionInput.tsx:36]: Having a separate component should be fine, though it seems the Input forwards the ref in https://github.com/grafana/grafana/blob/47f87171496f675bc44ddf839effaed1846bfa4e/packages/grafana-ui/src/components/Input/Input.tsx#L34 and the legacy input allows `inputRef` property https://github.com/grafana/grafana/blob/47f87171496f675bc44ddf839effaed1846bfa4e/packages/grafana-ui/src/components/Forms/Legacy/Input/Input.tsx#L85

**Review by ifrost — APPROVED** [review]
LGTM 👍

File: src/shared/components/QueryBuilder/domain/helpers/__tests__/queryToFilters.spec.ts (hunk 4 above)
Comment (grafakus) [reply src/shared/components/QueryBuilder/domain/helpers/__tests__/queryToFilters.spec.ts:104]: Yes, but only after finalizing https://github.com/grafana/explore-profiles/pull/122

Comment (ifrost) [reply src/shared/components/QueryBuilder/domain/helpers/__tests__/queryToFilters.spec.ts:104]: Yep, sorry I was looking at two PRs at the same time and got confused :D

File: src/shared/components/QueryBuilder/ui/inputs/SingleEditionInput.tsx (hunk 5 above)
Comment (grafakus) [reply src/shared/components/QueryBuilder/ui/inputs/SingleEditionInput.tsx:36]: Funny how I missed it. I'll have a look in the next PR. Thanks

**Pull request merged**